### PR TITLE
Implement stub for gigasecond

### PR DESCRIPTION
**Pull request by kytrinyx**
This one is going to need some tweaks to get the test suite looking reasonable.

Once I know what the test suite here should look like I can also generate a stub for the `meetup` problem.

I chose to not implement the first three tests from the canonical data, as they specify only a date rather than a datetime. The tests seem redundant, as we also have a test that specifies the same behavior, but with a datetime as an input.

I also chose not to implement the test for immutability, because from the docs it looks like the datetime functions return new values rather than mutating the existing value. If I'm wrong about that let me know and I'll update this to add the immutability test (I'll need to know how to check for object equality, or whatever the equivalent is called in Chapel).

### CODE REVIEW

**Comment by lucaferranti** [comment]
what would be the more pedagogical way of implementing this? Asking students to write their own `Datetime` record or guide them to the documentation of the built-in `datetime` module?

> (I'll need to know how to check for object equality, or whatever the equivalent is called in Chapel).

In chapel, records and classes come with a default equality check, so doing `test.assertEqual` without anything extra should work fine

**Comment by kytrinyx** [comment]
> what would be the more pedagogical way of implementing this?

I think guiding students to the built-in module is the right way to go here.

Are datetime records mutable? Or does the `add()` return a new record?

**Comment by lucaferranti** [comment]
> I think guiding students to the built-in module is the right way to go here.

yes, I now read the instructions more carefully and had a look at how it was done in other languages. For beginner exercise definitely better to point to the existing `Time` module.

> Are datetime records mutable?

TL;DR: Records can be mutated, but in this case it's definitely not something you can do by accident.

So in principle it is possible,
```chapel
use Time;
var a = new datetime(2022, 10, 08, hour=23, minute=59, second=37);
writeln(a);
a.chpl_date.chpl_year = 1997;
writeln(a);
```

```
2022-10-08T23:59:37
1997-10-08T23:59:37
```

but changing the fields directly is bad practice and in this case one would need to go peep the chapel source code to figure out the fields names (so extremely unlikely to happen). The public API methods would return a new instance (even `replace`).

Also, if the function is defined the normal way as `proc add(a: datetime) { }`, the value is passed as constant and to be edited and written back, one would need to know to manually change the definition to `proc add(inout a: datetime) `.

So the test is probably redundant, but could also be kept

**Comment by kytrinyx** [comment]
> So the test is probably redundant, but could also be kept

It sounds like we can safely skip it. Let's stick with just the two tests that are there already.

**Comment by lucaferranti** [comment]
> I chose to not implement the first three tests from the canonical data, as they specify only a date rather than a datetime.

yeah those tests would make more sense if one had to manually parse the date from a string. Since in this case the input is already given as `datetime` object, they are probably redundant. A datetime can also be constructed as `new datetime(2022, 10, 03)` in which case the time is midnight by default

**Comment by lucaferranti** [comment]
I did some small tweaks

- used variables in tests for readability
- changed the name of the function-under-test from `from` to `addGigasecond` (in the json-specs it seems to be called `add`)
- added `hints.md` pointing the student to the `Time` module documentation. I am not sure how this is displayed, does it need some more formatting (e.g. a `# Hints` header or similar)?

**Comment by kytrinyx** [comment]
Good call on changing `from` to `addGigasecond`. I think the header would get stripped from the hints if it were there, so this should be good.

I can't remember exactly how this shows up, and I don't have an example on hand at the moment. How do you feel about assuming that it'll be fine, and then tweaking later if we think it needs to be?

**Comment by lucaferranti** [comment]
> How do you feel about assuming that it'll be fine, and then tweaking later if we think it needs to be?

sounds great!

**Review by lucaferranti — APPROVED** [review]
